### PR TITLE
Api feature

### PR DESCRIPTION
**Pull request by Gbengacode**
### I made the following changes to this work

- I Added API calls
- I Added react thunk for asynchronous operation
- I refactored code to update both the react state and the API calls

### CODE REVIEW

File: src/pages/Books.js [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
       <ul>
         {books.map((book) => (
           <Book
-            key={book.id}
-            author={book.author}
+            key={book.item_id}
             title={book.title}
-            id={book.id}
+            item_id={book.item_id}
             category={book.category}
```
Comment (hiromataba) [inline src/pages/Books.js:22]: - [ ] Good job on  adding, deleting, and fetching your books to the API but when rendering your books there is a bit of issue, When I go to the categories page it works fine I see the message `under construction but when I navigate back to books they duplicate themselves but if I reload the page the duplication disappears, Kindly have a look at that video and fix the issue :point_down: 

https://user-images.githubusercontent.com/75126481/149944326-60b98c4f-8373-4eba-830b-be2c0373cb1c.MP4

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi :wave @Gbengacode , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight :tada: :champagne: 
- No linter errors :+1: 
- Nice PR description and title :+1: 
- Nice README file :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] [OPTIONAL] Your README looks professional however I think it would be better to add the screenshot of your app in it

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/Books.js (hunk 1 above)
Comment (Gbengacode) [reply src/pages/Books.js:22]: Thank you very much for the observation. The book reducer was returning a merged array previously

`return [...state, action.payload];`

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Gbengacode,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/Books.js (hunk 1 above)
Comment (hiromataba) [reply src/pages/Books.js:22]: You are welcome @Gbengacode

**Pull request merged**